ESSArch_Core/storage/backends/tests/test_disk.py

```diff
@@ -0,0 +1,277 @@
+        mock_storage_obj.get_full_path.return_value = "the_full_path"
+
+        disk_storage_backend.read(
+            storage_object=mock_storage_obj,
+            dst="some_dest"
+        )
+
+        mock_copy.assert_called_once_with("the_full_path", "some_dest", block_size=65536)
+
+    @mock.patch("ESSArch_Core.storage.models.StorageObject")
+    @mock.patch("ESSArch_Core.ip.models.InformationPackage")
+    @mock.patch("ESSArch_Core.storage.backends.disk.copy")
+    def test_read_container_default(self, mock_copy, mock_ip, mock_storage_obj):
+        disk_storage_backend = DiskStorageBackend()
+        full_path_to_src = "the/full/path/to/src"
+        container_path = os.path.join("container/path")
+
+        mock_storage_obj.container = True
+        mock_storage_obj.get_full_path.return_value = full_path_to_src
+        mock_storage_obj.storage_medium.storage_target.target = container_path
+        mock_storage_obj.ip = mock_ip
+        mock_ip.aic.pk = 1234
+
+        disk_storage_backend.read(
+            storage_object=mock_storage_obj,
+            dst="some_dest"
+        )
+
+        expected_copy_calls = [
+            mock.call(f"{full_path_to_src}.xml", "some_dest", block_size=65536),
+            mock.call(os.path.join(container_path, "1234.xml"), "some_dest", block_size=65536),
+            mock.call(f"{full_path_to_src}", "some_dest", block_size=65536)
+        ]
+
+        self.assertEqual(mock_copy.call_count, 3)
+        mock_copy.assert_has_calls(expected_copy_calls)
+
+    @mock.patch("ESSArch_Core.storage.models.StorageObject")
+    @mock.patch("ESSArch_Core.ip.models.InformationPackage")
+    @mock.patch("ESSArch_Core.storage.backends.disk.copy")
+    def test_read_container_no_xml(self, mock_copy, mock_ip, mock_storage_obj):
+        disk_storage_backend = DiskStorageBackend()
+        full_path_to_src = "the/full/path/to/src"
+        container_path = "container/path"
+
+        mock_storage_obj.container = True
+        mock_storage_obj.get_full_path.return_value = full_path_to_src
+        mock_storage_obj.storage_medium.storage_target.target = container_path
+        mock_storage_obj.ip = mock_ip
+        mock_ip.aic.pk = 1234
+
+        disk_storage_backend.read(
+            storage_object=mock_storage_obj,
+            dst="some_dest",
+            include_xml=False
+        )
+
+        mock_copy.assert_called_once_with(f"{full_path_to_src}", "some_dest", block_size=65536)
+
+    @mock.patch("ESSArch_Core.storage.backends.disk.DiskStorageBackend._extract")
+    @mock.patch("ESSArch_Core.storage.models.StorageObject")
+    @mock.patch("ESSArch_Core.ip.models.InformationPackage")
+    @mock.patch("ESSArch_Core.storage.backends.disk.copy")
+    def test_read_container_extract(self, mock_copy, mock_ip, mock_storage_obj, mock_extract):
+        disk_storage_backend = DiskStorageBackend()
+        full_path_to_src = "the/full/path/to/src"
+        container_path = os.path.join("container/path")
```

Comment:
`os.path.join` has no effect here

```suggestion
        container_path = "container/path"
```